test/e2e/v1beta1/e2e_test.go

```diff
@@ -706,4 +708,54 @@ var _ = Describe("For a Kubernetes cluster with Tekton and build installed", fun
 		})
 	})
 
+	Context("when tolerations are specified", Serial, func() {
+
+		BeforeEach(func() {
+			testID = generateTestID("tolerations")
+
+			// create the build definition
+			build = createBuild(
+				testBuild,
+				testID,
+				"test/data/v1beta1/build_buildah_tolerations_cr.yaml",
+			)
+
+			// Add a taint to both of the kind worker nodes
+			nodes, err := testBuild.GetNodes()
+			Expect(err).ToNot(HaveOccurred())
+			for _, node := range nodes.Items {
+				if node.Name == "kind-worker" || node.Name == "kind-worker2" {
+					taint := corev1.Taint{Key: "test-key", Value: "test-vaue", Effect: "NoSchedule"}
+					testBuild.AddNodeTaint(node.Name, &taint)
+				}
+			}
+		})
+
+		AfterEach(func() {
+			err := testBuild.RemoveNodeTaints("kind-worker")
+			Expect(err).ToNot(HaveOccurred())
+			err = testBuild.RemoveNodeTaints("kind-worker2")
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("successfully runs a build when it tolerates the node taint", func() {
+			buildRun, err = buildRunTestData(testBuild.Namespace, testID, "test/data/v1beta1/buildrun_buildah_tolerations_cr.yaml")
+			Expect(err).ToNot(HaveOccurred(), "Error retrieving buildrun test data")
+
+			buildRun = validateBuildRunToSucceed(testBuild, buildRun)
+			testBuild.ValidateImageDigest(buildRun)
+		})
+
+		It("fails to schedule when the build does not tolerate the node taint", func() {
+			buildRun, err = buildRunTestData(testBuild.Namespace, testID, "test/data/v1beta1/buildrun_buildah_tolerations_cr.yaml")
+			Expect(err).ToNot(HaveOccurred(), "Error retrieving buildrun test data")
+			buildRun.Spec.Tolerations[0].Key = "untolerated"
+
+			validateBuildRunToFail(testBuild, buildRun)
+			buildRun, err = testBuild.LookupBuildRun(types.NamespacedName{Name: buildRun.Name, Namespace: testBuild.Namespace})
+
+			Expect(buildRun.Status.FailureDetails.Message).To(Equal(shpgit.AuthPrompted.ToMessage()))
+			Expect(buildRun.Status.FailureDetails.Reason).To(Equal(shpgit.AuthPrompted.String()))
```

Comment:
Can you explain this. There is a BuildRun which cannot start because the tolerations do not match any node. Is not the result that the Pod is stuck in Pending, eventually the TaskRun and BuildRun time out. Why would there be a failure coming from the Git source step ?

Reply:
Ah, this was an oversight on my part. I would expect the Pending/timeout you mentioned instead. I'll change this.